content/blog/persisting-kubernetes-workloads-with-amazon-efscsi-volumes-using-pulumi-sdks/index.md

```diff
@@ -0,0 +1,397 @@
+
+The [AWS EFS CSI Driver](https://github.com/kubernetes-sigs/aws-efs-csi-driver) is a Kubernetes [SIG-AWS](https://github.com/kubernetes/community/tree/master/sig-aws) subproject. The AWS EFS CSI Driver triggers the creation of out-of-tree CSI volume plugins in any Kubernetes cluster. Thereafter Kubernetes storage classes, persistent volumes and claims can be used by a Kubernetes workload to access the EFS CSI volume plugin as before.
+
+In this blog, we will work through an example that shows how to use AWS EFS CSI storage components with Kubernetes workloads running on Amazon EKS worker nodes using Pulumi libraries ([EKS](https://github.com/pulumi/pulumi-eks), [AWS](https://github.com/pulumi/pulumi-aws), and [AWSX] (https://github.com/pulumi/pulumi-awsx/tree/master/nodejs/awsx)).
+
+## Step 1: Initialize Pulumi Project and Stack for your organization:
+[Install pulumi CLI](https://pulumi.io/quickstart/install.html?__hstc=194006706.92c420b2463a950f50b989da5e9a9de1.1559843842775.1560878104539.1560906741042.31&__hssc=194006706.2.1560906741042&__hsfp=3773980820) and set up your [AWS credentials](https://pulumi.io/quickstart/aws/setup.html?__hstc=194006706.92c420b2463a950f50b989da5e9a9de1.1559843842775.1560878104539.1560906741042.31&__hssc=194006706.2.1560906741042&__hsfp=3773980820). Initialize a new [Pulumi project](https://pulumi.io/reference/project.html?__hstc=194006706.92c420b2463a950f50b989da5e9a9de1.1559843842775.1560878104539.1560906741042.31&__hssc=194006706.2.1560906741042&__hsfp=3773980820) from available templates. We use `aws-typescript` template here to install all library dependencies.
+
+We will work with two Pulumi stacks in this example, one for the Amazon EKS cluster and AWS EFS CSI components caled k8sinfra. The other for the application and its storage class, persistent volume and persistent volume claim called app. The AWS EFS CSI (Container Storage Interface) is based on the initial [AWS EFS CSI Driver](https://github.com/kubernetes-sigs/aws-efs-csi-driver/) work done by Kubernetes [SIG-AWS](https://github.com/kubernetes/community/tree/master/sig-aws).
+
+```bash
+$ brew install pulumi # download pulumi CLI
+
+$ mkdir k8sinfra && cd k8sinfra
+
+$ pulumi new aws-typescript
+
+$ npm install --save @pulumi/kubernetes @pulumi/eks
+
+$ ls -la
+drwxr-xr-x   10 nishidavidson  staff    320 Jun 18 18:22 .
+drwxr-xr-x+ 102 nishidavidson  staff   3264 Jun 18 18:13 ..
+-rw-------    1 nishidavidson  staff     21 Jun 18 18:22 .gitignore
+-rw-r--r--    1 nishidavidson  staff     32 Jun 18 18:22 Pulumi.dev.yaml
+-rw-------    1 nishidavidson  staff     91 Jun 18 18:22 Pulumi.yaml
+-rw-------    1 nishidavidson  staff    273 Jun 18 18:22 index.ts
+drwxr-xr-x   95 nishidavidson  staff   3040 Jun 18 18:22 node_modules
+-rw-r--r--    1 nishidavidson  staff  50650 Jun 18 18:22 package-lock.json
+-rw-------    1 nishidavidson  staff    228 Jun 18 18:22 package.json
+-rw-------    1 nishidavidson  staff    522 Jun 18 18:22 tsconfig.json
+
```

Comment:
Nit: This blank line isn't needed, right?

Reply:
Done